x-pack/plugin/core/src/main/java/org/elasticsearch/license/XPackLicenseState.java

```diff
@@ -627,6 +650,22 @@ public boolean isAllowedByLicense(OperationMode minimumMode, boolean needActive)
         });
     }
 
+    /**
+     * Test whether current license expires in less than {@code LICENSE_EXPIRATION_WARNING_PERIOD}.
+     *
+     * @param now  Current time in milliseconds
+     *
+     * @return true if current license expires in less than {@code LICENSE_EXPIRATION_WARNING_PERIOD}, otherwise false
+     */
+    public boolean isLicenseExpiring(long now) {
```

Comment:
Given what this does, would it make sense to name it `shouldWarnAboutLicense()` ?